File: node/examples/content.ts
import 'dotenv/config'

import { NlpClient } from '@aristech-org/nlp-client'
// util is only used to print deep objects in the console
import util from 'util'

const auth = process.env.TOKEN && process.env.SECRET ? { token: process.env.TOKEN, secret: process.env.SECRET } : undefined

const client = new NlpClient({
  host: process.env.HOST,
  ssl: Boolean(auth) || Boolean(process.env.ROOT_CERT) || process.env.SSL === 'true',
  rootCert: process.env.ROOT_CERT,
  auth,
})

// Get matching content from the vector database
const response = await client.getContent({
  prompt: process.argv[2] || 'What are the lottery numbers?',
  threshold: 0.5,
  numResults: 3,
  projectId: process.env.PROJECT_ID,
})
console.log(util.inspect(response, { depth: null, colors: true }))

// Update content inside the vector database
/*
const updateResponse = await client.updateContent({
  intent: [{
    id: '123',
    topic: 'lottery-numbers',
    projectId: process.env.PROJECT_ID,
    inputs: [
      { uuid: '82a2133e-038e-4c83-aa26-de47ed386c55', input: 'What are the lottery numbers?' },
      { uuid: '2cc467ca-dc1a-4d33-92ff-8f73bb941fb9', input: 'Can you tell me the numbers from saturday?' },
    ],
    outputChat: [
      'The latest lottery numbers are {{lotto_numbers}}'
    ]
  }],
})
*/

// Remove content from the vector database
/*
const removeResponse = await client.removeContent({
  id: ['82a2133e-038e-4c83-aa26-de47ed386c55'],
  projectId: process.env.PROJECT_ID,
})
*/

File: python/examples/content.py
import sys

from aristech_nlp_client import NlpClient, GetContentRequest
from utils import host, auth_token, auth_secret, root_cert, ssl, project_id

prompt = sys.argv[1] if len(sys.argv) > 1 else "What are the lottery numbers?"

client = NlpClient(host=host, ssl=ssl, root_cert=root_cert, auth_token=auth_token, auth_secret=auth_secret)
responses = client.get_content(GetContentRequest(
    prompt=prompt,
    threshold=0.5,
    num_results=3,
    project_id=project_id,
))

try:
    print(next(responses))
    for response in responses:
        print(response) 
except StopIteration:
    print("No matching intents above the threshold.")

# To update content:
"""
from aristech_nlp_client import UpdateContentRequest, Intent, IntentInput
updateResponse = client.update_content(UpdateContentRequest(
    intent=[Intent(
        id="123",
        projectId=project_id,
        topic="lottery-numbers",
        inputs=[
            IntentInput(uuid="82a2133e-038e-4c83-aa26-de47ed386c55", input="What are the lottery numbers?"),
            IntentInput(uuid="2cc467ca-dc1a-4d33-92ff-8f73bb941fb9", input="Can you tell me the numbers from saturday?"),
        ],
        output_chat=[
            "The latest lottery numbers are {{lotto_numbers}}"
        ]
    )]
))
"""

# To remove content:
"""
from aristech_nlp_client import RemoveContentRequest
removeResponse = client.remove_content(RemoveContentRequest(
    id=["82a2133e-038e-4c83-aa26-de47ed386c55"],
    projectId=project_id,
))
"""
